### ISSUE TITLE
Add support for autoFocus to BreadcrumbItem and autoFocusCurrent to Breadcrumbs

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#2958]
# 🙋 Feature Request

`Breadcrumbs` should support the ability to `autoFocus` the current `BreadcrumbItem` link on page load or when the Breadcrumbs update.

`useBreadcrumbItem` makes use of `useLink`, which supports `autoFocus` via `useFocusable`. However, the `autoFocus` prop is not supported on `BreadcrumbItem`. 

## 🤔 Expected Behavior
1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/a93c1886c815cde4b39ead45d8768a4081daf1f2/docs/react-aria/useMenu.html
2. Tab to `AriaMenuOptions` link in the [API code example](https://reactspectrum.blob.core.windows.net/reactspectrum/a93c1886c815cde4b39ead45d8768a4081daf1f2/docs/react-aria/useMenu.html#api).
3. Press Enter to expand `AriaMenuOptions` popover.
4. BreadcrumbItem for the current location, `AriaMenuOptions`, should have focus 
5. Drill down to AriaMenuOptions > CollectionChildren > CollectionElement > SectionElement > SectionProps > ItemElement > ItemProps. The dropdown menu should display. After clicking each link to refresh the contents of the popover, the  BreadcrumbItem for the current location should automatically focus.
6. Open dropdown and arrow down to focus, then press space to activate CollectionElement.
CollectionElement BreadcrumbItem element should autoFocus if things are working correctly, otherwise focus will be lost to the document.body, and the next tab stop will be and element outside of the popover.

## 😯 Current Behavior

1. Open https://react-spectrum.adobe.com/react-aria/useMenu.html
2. Tab to `AriaMenuOptions` link in the [API code example](https://react-spectrum.adobe.com/react-aria/useMenu.html#api).
3. Press Enter to expand `AriaMenuOptions` popover.
4. Focus should be on the popover dialog itself and the BreadcrumbItem for the current location, `AriaMenuOptions`, will not have focus.
5. Tab to focus `CollectionChildren` link.
6. Press Enter to activate the link replacing the contents of the popover with API information for AriaMenuOptions > CollectionChildren.
7. The  BreadcrumbItem for the current location will not automatically focus, and focus will be lost to the `document.body`, relying on the browsers native cursor control to determine where focus should go next when navigating using the Tab key.

## 💁 Possible Solution

1. Expose `autoFocus` prop on `BreadcrumbItem`, and allow the current BreadcrumbItem to receive focus by adding `tabIndex="-1"` to the current item has `autoFocus === true`.
2. Add `autoFocusCurrent` prop to `Breadcrumbs`, which will in turn set `autoFocus={true}` for the `BreadcrumbItem` representing the current location.

This has been implemented in https://github.com/adobe/react-spectrum/pull/2749/commits/a93c1886c815cde4b39ead45d8768a4081daf1f2

## 🔦 Context

Trying to address concern in code comment at https://github.com/adobe/react-spectrum/pull/2749#pullrequestreview-907832387

> It's ok as a temporary solution for the docs, but I think we should really come up with a nicer solution. I don't think we should recommend manually mutating the DOM to users of React Spectrum.

## 💻 Examples

See https://reactspectrum.blob.core.windows.net/reactspectrum/a93c1886c815cde4b39ead45d8768a4081daf1f2/docs/react-aria/useMenu.html to test the expected behavior.

## 🧢 Your Company/Team

Adobe/Accessibility

## 🎁 Tracking Ticket (optional)

Related to #1961 and #2764.
